xml/admin_backup.xml

```diff
@@ -0,0 +1,394 @@
+    <step>
+     <para>
+      Change to the <filename>/var/tmp</filename> directory and package
+      important data into a tar file archive:
+     </para>
+<screen>&prompt.root;<command>
+tar czf <replaceable>backup-admin-node-$( date +'%Y-%m-%d' ).tgz</replaceable> -C / \
+    var/lib/misc/salt-master-credentials \
+    usr/share/salt \
+    etc/pki usr/share/caasp-container-manifests \
+    etc/salt/pki \
+    etc/caasp \
+    var/lib/misc/salt/cache \
+    etc/caasp/haproxy \
+    var/lib/mysql \
+    var/lib/misc/infra-secrets \
+    var/lib/misc/ldap \
+    var/lib/misc/ldap-config \
+    var/lib/misc/ssh-public-key \
+    var/lib/misc/infra-secrets \
+    etc/salt/pillar \
+    var/lib/misc/velum-secrets</command>
+ </screen>
+    </step>
+    <step>
+     <para>
+      Then, export the MariaDB database and archive it:
+     </para>
+ <screen>&prompt.root;<command>docker exec $(docker ps -qf name="velum-mariadb") mysqldump \
+ --password=$(cat /var/lib/misc/infra-secrets/mariadb-root-password) \
+ velum_production | gzip > <replaceable>/var/tmp/db-backup-[$(date+'%Y-%m-%d')].sql.gz</replaceable></command>
+ </screen>
+    </step>
+    <step>
+     <para>
+      Finally, copy the created files <filename>backup-admin-node-[date].tgz</filename>
+      and <filename>/var/tmp/db-backup-[date].sql.gz</filename> to
+      a location of your choosing; in accordance with your best practices for
+      backup files.
+     </para>
+     <important>
+      <title>Sensitive Data</title>
+      <para>
+       Please note that the data backup files contain senstive information that
+       allow full cluster access and must be handled securely.
+      </para>
+     </important>
+    </step>
+   </procedure>
+  </sect2>
+
+  <sect2 xml:id="sec.admin.backup.admin.restore">
+   <title>Restore Procedure</title>
+   <procedure>
+    <step>
+     <para>
+      Set up a new machine for the &admin_node; by following
+      <xref linkend="sec.deploy.nodes.admin_install" />.
+     </para>
+     <para>
+      You must be able to access &dashboard; indicating all containers are
+      correctly started. There is, however, no need to complete the
+      configuration steps in &dashboard;.
+     </para>
+    </step>
+    <step>
+     <para>
+      Copy the <filename>backup-admin-node-[date].tgz</filename> and
+      <filename>db-backup-[date].sql.gz</filename> of the most recent backups
+      onto the newly created &admin_node; machine. Make sure to replace
+      <literal>[date]</literal> with the actual date subsituted during the
+      backup procedure.
+     </para>
+    </step>
+    <step>
+     <para>
+      SSH into the new &admin_node; machine and switch to the location where
+      you have stored the backup files.
+     </para>
+    </step>
+    <step>
+     <para>
+      Run the following command to unpack the data backup:
+     </para>
+<screen>&prompt.root;<command>tar xzf <replaceable>backup-admin-node-[date].tgz</replaceable> -C /</command></screen>
```

Comment:
This approach doesn't work on RO filesystem, first you have to switch fs to RW mode somehow (which is most likely not supported) or call `transactional-update shell` and do it there. I'm trying to find out working method.

```
admin # tar xzf backup-admin-node-2019-02-26.tgz -C /
tar: usr/share/salt/kubernetes/config/master.d/50-api.conf: Cannot open: File exists
tar: usr/share/salt/kubernetes/config/master.d/50-master.conf: Cannot open: File exists
tar: usr/share/salt/kubernetes/config/master.d/50-peer.conf: Cannot open: File exists
tar: usr/share/salt/kubernetes/config/master.d/50-reactor.conf: Cannot open: File exists
tar: usr/share/salt/kubernetes/config/master.d/50-returner.conf: Cannot open: File exists
tar: usr/share/salt/kubernetes/config/master.d: Cannot utime: Read-only file system
tar: usr/share/salt/kubernetes/config/master.d: Cannot change ownership to uid 0, gid 0: Read-only file system
tar: usr/share/salt/kubernetes/config: Cannot utime: Read-only file system
tar: usr/share/salt/kubernetes/config: Cannot change ownership to uid 0, gid 0: Read-only file system
```

Reply:
Ok, running `transactional-update shell` (t-u shell) and extracting the backup-admin.tgz file within the shell do the trick.

So please modify this step to have this meaning:
* move the file backup-admin.tgz to /root - because /var/tmp is not mounted in the t-u shell env
* run `transactional-update shell`
* `cd /root` and perform `tar zxf backup-admin.tgz -C /`
* type `exit` to leave t-u shell env
* call `systemctl reboot` to take a new btrfs snapshot with restored data into account

Continue with db restore - `t-u shell` is not needed for that...

Reply:
this is not needed `/var/tmp` should be writable and is probably why it was not avalible within the transactional-update shell

Reply:
@Klaven sorry, but I'm not getting your point - we need to switch to `t-u shell` to do `tar xz ...` because of RO fs but in this shell `/var/tmp/` is not mounted so we need to move the backup to /root (or another location visible under `t-u shell`)

Reply:
We clarified with @Klaven that `t-u shell` is really not needed...  As we restore only into writable dirs like /etc/ and the rest is not interesting for the backup, but anyway the command will produce lot of garbage output/errors

@r0ckarong you can ignore this thread